libmamba/src/core/package_handling.cpp

```diff
@@ -354,14 +356,53 @@ namespace mamba
         LOG_INFO << "Extracting " << file << " to " << destination;
         extraction_guard g(destination);
 
+        struct archive* a;
+        int r;
+
+        a = archive_read_new();
+        archive_read_support_format_tar(a);
+        archive_read_support_format_zip(a);
+        archive_read_support_filter_all(a);
+
+        auto lock = LockFile(file);
+        r = archive_read_open_filename(a, file.string().c_str(), 10240);
+
+        if (r != ARCHIVE_OK)
+        {
+            LOG_ERROR << "Error opening archive: " << archive_error_string(a);
+            throw std::runtime_error(file.string() + " : Could not open archive for reading.");
+        }
+
+        stream_extract_archive(a, destination);
+
+        archive_read_close(a);
+        archive_read_free(a);
+    }
+
+    namespace
+    {
+        struct conda_extract_context
+        {
+            conda_extract_context(struct archive* source)
```

Comment:
`struct` is unnecessary here
Also make this constructor `explicit` otherwise you will be able to pass ints to functions taking this.

Reply:
BTW is it ok if `source` is `nullptr`? If not consider adding an assertion in the body.

Reply:
I fixed this by taking the raii object as input now.